node/core/prospective-parachains/src/fragment_tree.rs

```diff
@@ -665,46 +665,46 @@ impl FragmentTree {
 				};
 
 			let parent_head_hash = candidate.parent_head_data_hash();
-			if parent_head_hash == child_constraints.required_parent.hash() {
-				// We do additional checks for complete candidates.
-				if let HypotheticalCandidate::Complete {
-					ref receipt,
-					ref persisted_validation_data,
-				} = candidate
-				{
-					let prospective_candidate = ProspectiveCandidate {
-						commitments: Cow::Borrowed(&receipt.commitments),
-						collator: receipt.descriptor().collator.clone(),
-						collator_signature: receipt.descriptor().signature.clone(),
-						persisted_validation_data: persisted_validation_data.as_ref().clone(),
-						pov_hash: receipt.descriptor().pov_hash,
-						validation_code_hash: receipt.descriptor().validation_code_hash,
-					};
+			if parent_head_hash != child_constraints.required_parent.hash() {
+				continue
+			}
```

Comment:
Changed this check to `continue`-style to be consistent with above checks (and it's easier to read). Rest of the diff here is due to indentation/formatting.